src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -75,12 +74,21 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
             this.clearAllRegions();
             this.setState({
                 currentAsset: this.props.selectedAsset,
-                selectedRegions: [],
             });
         }
 
         if (this.props.selectionMode !== prevProps.selectionMode) {
-            this.editor.AS.setSelectionMode(this.props.selectionMode, null);
+            let  options = null;
+            if (this.props.selectionMode === SelectionMode.COPYRECT) {
+                const selectedRegion = this.getSelectedRegions(); // this.state.selectedRegions[0];
+                if (selectedRegion) {
+                    options = {template: new Rect(selectedRegion[0].boundingBox.width,
+                                                selectedRegion[0].boundingBox.height)};
+                } else {
+                    throw new Error("No region selected to copy");
```

Comment:
If no template is provided, CanvasTools will use a default one (20x20 square). So such use case is not considered as an error.

Reply:
Just in case you would like to cover it with a test: set the AreaSelector.DefaultTemplateSize first to make it controllable. It defaults to `new Rect(20, 20)`